velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -223,11 +256,37 @@ dwio::common::StripeProgress getStripeProgress(
  *
  * This method assumes each input `ColumnarBatch` have same schema.
  */
-void Writer::write(const VectorPtr& data) {
+void Writer::write(const VectorPtr& dataToWrite) {
+  VELOX_USER_CHECK(
+      dataToWrite->type()->equivalent(*schema_),
+      "The file schema type should be equal with the input rowvector type.");
+  auto rowVector =
+      std::dynamic_pointer_cast<facebook::velox::RowVector>(dataToWrite);
+  VELOX_CHECK_NULL(rowVector->nulls());
+
+  // Update the schema for the complex type.
+  const auto& children = rowVector->children();
+  const auto size = schema_->size();
+  std::vector<VectorPtr> newChildren = children;
+  for (auto i = 0; i < size; i++) {
+    auto columnType = schema_->childAt(i);
+    bool isComplexType = std::dynamic_pointer_cast<const RowType>(columnType) ||
+        std::dynamic_pointer_cast<const MapType>(columnType) ||
+        std::dynamic_pointer_cast<const ArrayType>(columnType);
+
+    if (isComplexType) {
```

Comment:
Shouldn't this be done recursively? Will with work with complex typed nested 2, 3, or more levels deep?